### PR TITLE
shrshinde/isSupported-location-chat-conversation-files-check

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
- Adding isSupported check for location capability
- Adding unit test cases

### CODE REVIEW

File: packages/teams-js/src/public/location.ts [hunk 1]
```diff
@@ -77,6 +78,9 @@ export namespace location {
       if (!props) {
         throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
       }
+      if (!isSupported()) {
+        throw new Error(errorNotSupportedOnPlatform);
```
Comment (erinha) [inline packages/teams-js/src/public/location.ts:82]: We should be throwing SdkError not Error. This should apply to all places in the code where we check `isSupported`.

Comment (shrshindeMSFT) [reply packages/teams-js/src/public/location.ts:82]: Okay, I'll make those changes everywhere.

Comment (shrshindeMSFT) [reply packages/teams-js/src/public/location.ts]: Resolved in the latest commit.

**Review by erinha — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/location.ts [hunk 2]
```diff
@@ -77,6 +78,9 @@ export namespace location {
       if (!props) {
         throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
       }
+      if (!isSupported()) {
+        resolve(sendAndHandleError('location.getLocation', errorNotSupportedOnPlatform));
+      }
```
[the remark this answers is not in the document]
Comment (jadahiya-MSFT) [reply packages/teams-js/src/public/location.ts:83]: Above we have 
```
if(!props)  {

throw {errorCode: ErrorCode.INVALID_ARGUMENTS};

}
```

Can we not just use this structure?

[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/location.ts]: Fixed in the latest commit.

File: packages/teams-js/test/private/conversations.spec.ts [hunk 3]
```diff
@@ -114,7 +112,12 @@ describe('conversations', () => {
     it('closeConversation should throw error if conversation capability is not supported in runtime config', async () => {
       await utils.initializeWithContext(FrameContexts.content);
       utils.setRuntimeConfig({ apiVersion: 1, supports: { chat: {} } });
-      expect(() => conversations.closeConversation()).toThrowError(errorNotSupportedOnPlatform);
+      try {
+        conversations.closeConversation();
+      } catch (e) {
+        expect(e).toEqual(errorNotSupportedOnPlatform);
+      }
+      // expect().toMatchObject(errorNotSupportedOnPlatform);
```
Comment (jadahiya-MSFT) [inline packages/teams-js/test/private/conversations.spec.ts:120]: Is this intentional?

Comment (shrshindeMSFT) [reply packages/teams-js/test/private/conversations.spec.ts:120]: Forgot to remove those

**Review by jadahiya-MSFT — DISMISSED** [review]

File: packages/teams-js/test/private/files.spec.ts [hunk 4]
```diff
@@ -398,21 +399,23 @@ describe('files', () => {
       await utils.initializeWithContext('content');
```
Comment (erinha) [inline packages/teams-js/test/private/files.spec.ts:399]: Can you please add `expect.assertions(1)` (or whatever the correct number is) to ensure that the test validates the `expect` in the catch block?

File: packages/teams-js/test/private/files.spec.ts [hunk 5]
```diff
@@ -398,21 +399,23 @@ describe('files', () => {
...
-      ).toThrowError(errorNotSupportedOnPlatform);
+        );
+      } catch (e) {
+        expect(e).toEqual(errorNotSupportedOnPlatform);
+      }
     });
   });
   describe('openFilePreview', () => {
     it('openFilePreview should throw error when files capability is not supported', async () => {
       await utils.initializeWithContext('content');
```
Comment (erinha) [inline packages/teams-js/test/private/files.spec.ts:415]: Please add `expect.assertions(1)` to validate `expect` in catch block is called.

File: packages/teams-js/test/private/files.spec.ts [hunk 6]
```diff
@@ -570,8 +575,11 @@ describe('files', () => {
     it('getFileDownloads should throw error when files capability is not supported', async () => {
       await utils.initializeWithContext('content');
```
Comment (erinha) [inline packages/teams-js/test/private/files.spec.ts:576]: Please add `expect.assertions(1)` to validate catch block is executed

File: packages/teams-js/test/private/files.spec.ts [hunk 7]
```diff
@@ -619,7 +627,11 @@ describe('files', () => {
       it('getFileDownloadsV2 should throw error when files capability is not supported', async () => {
         await utils.initializeWithContext('content');
```
Comment (erinha) [inline packages/teams-js/test/private/files.spec.ts:628]: `expect.assertions(1)`

File: packages/teams-js/test/private/files.spec.ts [hunk 8]
```diff
@@ -653,8 +665,11 @@ describe('files', () => {
     it('openDownloadFolder should throw error when files capability is not supported', async () => {
       await utils.initializeWithContext('content');
```
Comment (erinha) [inline packages/teams-js/test/private/files.spec.ts:666]: `expect.assertions(1)`

**Review by erinha — CHANGES_REQUESTED** [review]
Looks great, just want to make sure catch blocks are executed in some of the test cases, using `expect.assertions`.

File: packages/teams-js/test/private/conversations.spec.ts [hunk 9]
```diff
@@ -114,7 +112,11 @@ describe('conversations', () => {
     it('closeConversation should throw error if conversation capability is not supported in runtime config', async () => {
       await utils.initializeWithContext(FrameContexts.content);
```
Comment (erinha) [inline packages/teams-js/test/private/conversations.spec.ts:113]: Looks like I missed a place where we need `expect.assertions`

Comment (shrshindeMSFT) [reply packages/teams-js/test/private/conversations.spec.ts:113]: Okay I'll add it every where

**Review by erinha — CHANGES_REQUESTED** [review]
One more expect.assertions  needed

**Review by erinha — APPROVED** [review]
:shipit:

**Review by jadahiya-MSFT — APPROVED** [review]

**Pull request merged**